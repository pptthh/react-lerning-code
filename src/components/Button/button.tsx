import * as React from 'react';

export interface IButton {
    name?: string;
    label: string;
    tooltip?: string;
    className?: string;
    disabled?: boolean;
    onClick: React.MouseEventHandler<HTMLButtonElement>;
}

const Button = ({disabled=false, name, label, onClick, className, tooltip=''}: IButton) =>
    <button
        className={className}
        name={name}
        title={tooltip}
        onClick={onClick}
        disabled={disabled}
        >{label}</button>;

export default Button;
