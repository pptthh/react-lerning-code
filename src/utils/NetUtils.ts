import { EVENT_VALUE } from '.';

const netUtils = {

    getJsonResponse : (response: Response) => response.json(),

    checkStatus: (response: Response) => {
        if (Number.isInteger(response.status)) {
            return response;
        }
        throw Error('missing status code');
    },

    MOVIES_URL: 'http://react-cdp-api.herokuapp.com/movies',

    setUrlPath: (path: string) => {
        setTimeout(() => {
            !location.hash ?
                location.pathname = '/' + path :
                location.hash = '#/' + path;
        }, 0);
    },
};

export default netUtils;
