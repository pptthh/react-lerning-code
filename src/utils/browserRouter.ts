// tslint:disable-next-line
const BrowserRouter = require('react-router-dom').BrowserRouter;

export default BrowserRouter;
