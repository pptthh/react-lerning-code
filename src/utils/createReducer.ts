import { IActions } from "../scenes/Root/rootActions";

export interface ICase<T> {
    state: T;
    payload?: unknown;
}

export interface ISwitch<T> {
    [s: string]: ({state, payload}: ICase<T>) => T;
}

export const createReducer =
    <T>(Switch: ISwitch<T>, stateInit: T) => 
        (state: T = stateInit, {type, payload}: IActions<unknown>): T => (
            Switch.hasOwnProperty(type) ? Switch[type]({state, payload}) :
                state
        );


export default createReducer;
