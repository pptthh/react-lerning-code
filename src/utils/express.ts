process.env.NODE_SERVER = 'true';

// tslint:disable-next-line
const Express = require('express');
export default Express;
