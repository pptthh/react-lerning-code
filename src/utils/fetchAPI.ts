// tslint:disable-next-line
const fetchAPI = require('node-fetch');
export default fetchAPI;
