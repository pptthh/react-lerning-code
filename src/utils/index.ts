// tslint:disable-next-line
export const NOOP = () => {};

const c = console;
export const LOG = c.log;

export const throwError = (message?: string): any => { throw new Error(message); };

export const ECHO = <T>(value:T) => value;