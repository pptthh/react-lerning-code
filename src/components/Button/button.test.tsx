import { configure, render, shallow } from 'enzyme';
import * as ReactSixteenAdapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import * as React from 'react';
import { NOOP } from '../../utils';
import Button from './button';

configure({adapter: new ReactSixteenAdapter()});

describe('describe Button', () => {
    it('renders correctly with Label only', () => {
        const tree = render(<Button label='simple text22' onClick={NOOP}/>);

        expect(toJson(tree)).toMatchSnapshot();
    });

    it('shallow renders correctly with params', () => {
        let clickCounter = 0;
        const tree = shallow(
            <Button name='name' label='btn' onClick={() => {clickCounter++; }}>
                renders correctly with params
            </Button>);

        expect(toJson(tree)).toMatchSnapshot();
        tree.simulate('click');
        expect(clickCounter).toEqual(1);
    });
});
