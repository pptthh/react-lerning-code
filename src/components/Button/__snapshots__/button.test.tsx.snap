// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`describe Button renders correctly with Label only 1`] = `
<button
  title=""
>
  simple text22
</button>
`;

exports[`describe Button shallow renders correctly with params 1`] = `
<button
  disabled={false}
  name="name"
  onClick={[Function]}
  title=""
>
  btn
</button>
`;
